pkg/model/loader.go

```diff
@@ -146,11 +164,22 @@ func (ml *ModelLoader) CheckIsLoaded(s string) *Model {
 	}
 
 	log.Debug().Msgf("Model already loaded in memory: %s", s)
-	alive, err := m.GRPC(false, ml.wd).HealthCheck(context.Background())
+	client := m.GRPC(false, ml.wd)
+
+	log.Debug().Msgf("Checking model availability (%s)", s)
+	cTimeout, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
this PR also sets a timeout to the context in case calling HealthCheck to the backend stalls